lib/lambda_ethereum_consensus/state_transition/operations.ex

```diff
@@ -4,7 +4,127 @@ defmodule LambdaEthereumConsensus.StateTransition.Operations do
   """
 
   alias LambdaEthereumConsensus.StateTransition.{Accessors, Misc, Mutators, Predicates}
-  alias SszTypes.{Attestation, BeaconState, ExecutionPayload, Validator, Withdrawal}
+  alias LambdaEthereumConsensus.Utils.BitVector
+
+  alias SszTypes.{
+    Attestation,
+    BeaconState,
+    ExecutionPayload,
+    SyncAggregate,
+    Validator,
+    Withdrawal
+  }
+
+  @spec process_sync_aggregate(BeaconState.t(), SyncAggregate.t()) ::
+          {:ok, BeaconState.t()} | {:error, String.t()}
+  def process_sync_aggregate(
+        %BeaconState{
+          slot: slot,
+          current_sync_committee: current_sync_committee,
+          validators: validators
+        } = state,
+        %SyncAggregate{
+          sync_committee_bits: sync_committee_bits,
+          sync_committee_signature: sync_committee_signature
+        }
+      ) do
+    # Verify sync committee aggregate signature signing over the previous slot block root
+    committee_pubkeys = current_sync_committee.pubkeys
+
+    # TODO: Change bitvectors to be in little-endian instead of converting manually
+    sync_committee_bits_as_num = sync_committee_bits |> :binary.decode_unsigned()
+
+    sync_committee_bits =
+      <<sync_committee_bits_as_num::unsigned-integer-little-size(bit_size(sync_committee_bits))>>
+
+    participant_pubkeys =
+      Enum.with_index(committee_pubkeys)
+      |> Enum.filter(fn {_, index} -> BitVector.set?(sync_committee_bits, index) end)
+      |> Enum.map(fn {public_key, _} -> public_key end)
+
+    previous_slot = max(slot, 1) - 1
+    epoch = Misc.compute_epoch_at_slot(previous_slot)
+    domain = Accessors.get_domain(state, Constants.domain_sync_committee(), epoch)
+
+    with {:ok, block_root} <- Accessors.get_block_root_at_slot(state, previous_slot),
+         signing_root <- Misc.compute_signing_root(block_root, domain),
+         {:ok, true} <-
+           Bls.eth_fast_aggregate_verify(
+             participant_pubkeys,
+             signing_root,
+             sync_committee_signature
+           ) do
+      # Compute participant and proposer rewards
+      total_active_increments =
+        div(
+          Accessors.get_total_active_balance(state),
+          ChainSpec.get("EFFECTIVE_BALANCE_INCREMENT")
+        )
+
+      total_base_rewards =
+        Accessors.get_base_reward_per_increment(state) * total_active_increments
+
+      max_participant_rewards =
+        (total_base_rewards * Constants.sync_reward_weight())
+        |> div(Constants.weight_denominator())
+        |> div(ChainSpec.get("SLOTS_PER_EPOCH"))
+
+      participant_reward = div(max_participant_rewards, ChainSpec.get("SYNC_COMMITTEE_SIZE"))
+
+      proposer_reward =
+        (participant_reward * Constants.proposer_weight())
+        |> div(Constants.weight_denominator() - Constants.proposer_weight())
```

Comment:
We could move all this into a helper that returns both the proposer and participant rewards.

Reply:
Broke it up a bit